quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java

```diff
@@ -234,17 +236,43 @@ void configureProfile(KeycloakRecorder recorder) {
         recorder.configureProfile(profile.getName(), profile.getFeatures());
     }
 
+    @Record(ExecutionTime.STATIC_INIT)
+    @BuildStep
+    @Consume(ConfigBuildItem.class)
+    void configureRedirectForRootPath(BuildProducer<RouteBuildItem> routes,
+                                      HttpRootPathBuildItem httpRootPathBuildItem,
+                                      KeycloakRecorder recorder) {
+        var relativePath = Configuration.getOptionalKcValue(HttpOptions.HTTP_RELATIVE_PATH)
+                .filter(StringUtil::isNotBlank)
+                .filter(f -> !f.equals("/"));
+        if (relativePath.isEmpty()) return;
+
+        routes.produce(httpRootPathBuildItem.routeBuilder()
+                .route("/")
+                .handler(recorder.getRedirectHandler(relativePath.get()))
+                .build());
```

Comment:
Just a nit, we can avoid the explicit if / return:

```suggestion
        Configuration.getOptionalKcValue(HttpOptions.HTTP_RELATIVE_PATH)
                .filter(StringUtil::isNotBlank)
                .filter(f -> !f.equals("/"))
                .ifPresent(ignored ->  routes.produce(httpRootPathBuildItem.routeBuilder()
                        .route("/")
                        .handler(recorder.getRedirectHandler(relativePath.get()))
                        .build()));
```

Reply:
Ok works for me. Thanks